tool/m4/ruby_default_arch.m4

```diff
@@ -10,7 +10,7 @@ AS_CASE([$1:"$host_cpu"],
     [[i[3-6]86]:x86_64], [ARCH_FLAG=-m32],
     [ppc64:ppc*],        [ARCH_FLAG=-m64],
     [ppc*:ppc64],        [ARCH_FLAG=-m32],
-    AS_CASE([$build_os],
+    AS_CASE([$target_os],
```

Comment:
I tried this, but it's still not working yet. I wonder if `$target_os` has to be passed in as an argument as `$host_cpu` is.

Reply:
Changed to detect a possible option.